### PR TITLE
Add support for ServiceNow query with multiple args

### PR DESCRIPTION
**Pull request by DinaMeylakh**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/30284

## Description
Bug fix & added unittests

## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Review by teizenman — CHANGES_REQUESTED** [review]
Nicely done

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 1]
```diff
@@ -773,9 +774,16 @@ def query(self, table_name: str, sys_param_limit: str, sys_param_offset: str, sy
...
         """
+
         query_params = {'sysparm_limit': sys_param_limit, 'sysparm_offset': sys_param_offset}
         if sys_param_query:
-            query_params['sysparm_query'] = sys_param_query
+            if '&' in sys_param_query:
+                query_params['sysparm_query'] = []  # type: ignore
+                sysparm_query_args = parse.parse_qsl(sys_param_query)
+                for arg in sysparm_query_args:
+                    query_params['sysparm_query'].append(parse.urlencode([arg]))  # type: ignore
```
[the remark this answers is not in the document]
Comment (DinaMeylakh) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py]: Done.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py [hunk 2]
```diff
@@ -663,3 +663,55 @@ def test_update_remote_data_sc_task(mocker):
...
+                    'incident_name')
+    query = "assigned_to=123&active=true"
+    table_name = "sc_task"
+    requests_mock.request('GET', f'{url}table/{table_name}?sysparm_limit=50&sysparm_offset=0&'
+                                 'sysparm_query=assigned_to%3D123&sysparm_query=active%3Dtrue',
+                          json=RESPONSE_TICKET_ASSIGNED)
+    args = {'limit': "50", 'query': query, 'table_name': table_name}
+    human_readable, entry_context, result, bol = query_table_command(client, args)
+
+    assert result == RESPONSE_TICKET_ASSIGNED
```
[the remark this answers is not in the document]
Comment (DinaMeylakh) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py]: Done.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 3]
```diff
@@ -421,6 +422,26 @@ def split_fields(fields: str = '') -> dict:
     return dic_fields
 
 
+def handle_ampersands_in_query(query):
```
Comment (teizenman) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:425]: Maybe use a name like `build_query_for...`

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py [hunk 4]
```diff
@@ -663,3 +663,47 @@ def test_update_remote_data_sc_task(mocker):
     mocker.patch('ServiceNowv2.get_ticket_fields', side_effect=ticket_fields)
     mocker.patch.object(client, 'update', side_effect=update_ticket)
     update_remote_system_command(client, args, params)
+
+
+def test_handle_ampersands_in_query():
```
Comment (teizenman) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py:668]: add more cases + edge cases

**Review by teizenman — CHANGES_REQUESTED** [review]

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py (hunk 4 above)
Comment (DinaMeylakh) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py]: The only other one I can think of is a recursive '&' usage like "banana=3&user=admin=true&id=5" but I don't know if requests.requests supports it nor if there is a use/support for it in ServiceNow queries. Also it is out of scope of the bug and I'm not sure its worth the extra effort. 

If you have another use case in mind that I didn't think of i'd be happy to hear

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py (hunk 3 above)
Comment (DinaMeylakh) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py]: 🥔 🥔

**Review by teizenman — APPROVED** [review]

**Pull request merged**